weave-js/vite-plugin-block-cjs.ts

```diff
@@ -1,6 +1,8 @@
 import * as path from 'path';
 import pkg from 'lodash';
-import {visit} from 'ast-types';
+import {visit, Type} from 'ast-types';
+
+Type.def('StaticBlock').finalize();
```

Comment:
What's this do?

Reply:
The plugin in this file walks the trees of our source files looking for potentially problematic inputs. When I upgraded to Vite 5, I started getting an error that the walker was encountering an [unidentified syntax node type called `StaticBlock`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks) -- I'm guessing that either:

* some transitive dependency was updated, and the new version has added use of static blocks, or
* the parser being used by Vite (which our plugin implicitly uses) was upgraded to a new version that parses these blocks as a distinct type, rather than some generic language construct

Either way, the `ast-types` walker didn't know what to do with it, and there wasn't a new version of `ast-types` to upgrade to. There's probably some other library that's supplanted it, but since I couldn't figure out what to use with a quick search, I just went to the `ast-types` documentation and found how to register new node types.

Normally you'd have to add a bunch of handlers to tell it what to, like... actually _do_ with these new nodes. But since we're only looking for imports and imports should never be in static blocks, I figured it was safe to do this -- basically just telling the walker to ignore them.